sql-plugin/src/main/330+/scala/com/nvidia/spark/rapids/shims/AnsiUtil.scala

```diff
@@ -58,15 +60,19 @@ object AnsiUtil extends Arm {
 
     withResource(doubleInput.isNan) { hasNan =>
       if (BoolUtils.isAnyValidTrue(hasNan)) {
-        throwDateTimeException
+        withResource(FloatUtils.getNanScalar(doubleInput.getType)) { nan =>
+          throwSparkDateTimeException(nan.getDouble.toString)
+        }
       }
     }
 
     // check nan
     withResource(FloatUtils.getInfinityVector(doubleInput.getType)) { inf =>
       withResource(doubleInput.contains(inf)) { hasInf =>
         if (BoolUtils.isAnyValidTrue(hasInf)) {
-          throwDateTimeException
+          withResource(FloatUtils.getPositiveInfinityScalar(doubleInput.getType)) { infinity =>
+            throwSparkDateTimeException(infinity.getDouble.toString)
```

Comment:
just Pass string "Infinity" and also add a comment here that we don't specify when -Infinity